examples/chef/common/clusters/channel/ChannelManager.cpp

```diff
@@ -0,0 +1,343 @@
+    const chip::Optional<uint32_t> & endTime,
+    const chip::Optional<chip::app::DataModel::DecodableList<ChannelInfoType>> & channelList,
+    const chip::Optional<PageTokenType> & pageToken, const chip::Optional<chip::BitMask<RecordingFlagBitmap>> & recordingFlag,
+    const chip::Optional<chip::app::DataModel::DecodableList<AdditionalInfoType>> & externalIdList,
+    const chip::Optional<chip::ByteSpan> & data)
+{
+
+    // 1. Decode received parameters
+    // 2. Perform search
+    // 3. Return results
+
+    // PageTokenType paging;
+    // paging.limit  = MakeOptional(static_cast<uint16_t>(10));
+    // paging.after  = MakeOptional(chip::CharSpan::fromCharString("after-token"));
+    // paging.before = MakeOptional(chip::CharSpan::fromCharString("before-token"));
+
+    // ChannelPagingStructType channelPaging;
+    // channelPaging.nextToken = MakeOptional<DataModel::Nullable<Structs::PageTokenStruct::Type>>(paging);
+
+    std::vector<ProgramType> matches;
+    for (auto const & program : mPrograms)
+    {
+        if (startTime.ValueOr(0) > program.startTime)
+        {
+            continue;
+        }
+        if (endTime.HasValue() && endTime.ValueOr(0) < program.endTime)
+        {
+            continue;
+        }
+        if (channelList.HasValue())
+        {
+            auto iter     = channelList.Value().begin();
+            bool match    = false;
+            int listCount = 0;
+            while (iter.Next() && !match)
+            {
+                listCount++;
+                auto & channel = iter.GetValue();
+                if (channel.minorNumber != program.channel.minorNumber || channel.majorNumber != program.channel.majorNumber)
+                {
+                    continue;
+                }
+                // this sample code does not currently check OTT
+                match = true;
+            }
+            if (!match && listCount > 0)
+            {
+                continue;
+            }
+        }
+        // this sample code does not currently filter on external id list
+        matches.push_back(program);
+    }
+
+    ProgramGuideResponseType response;
+    response.programList = DataModel::List<const ProgramType>(matches.data(), matches.size());
+    helper.Success(response);
+}
+
+bool ChannelManager::HandleRecordProgram(const chip::CharSpan & programIdentifier, bool shouldRecordSeries,
+                                         const DataModel::DecodableList<AdditionalInfo> & externalIdList,
+                                         const chip::ByteSpan & data)
+{
+    // Start recording
+    std::string idString(programIdentifier.data(), programIdentifier.size());
+    for (auto & program : mPrograms)
+    {
+        std::string nextIdString(program.identifier.data(), program.identifier.size());
+        if (nextIdString == idString)
+        {
+            program.recordingFlag = MakeOptional(static_cast<uint32_t>(shouldRecordSeries ? 2 : 1));
```

Comment:
what are 2 or 1 ? please add constants for them

Reply:
Modified to 
```
program.recordingFlag = MakeOptional(static_cast<uint32_t>(shouldRecordSeries ? RecordingFlagBitmap::kRecordSeries : RecordingFlagBitmap::kScheduled));
```